lib/src/components/NxDonutChart/NxDonutChart.tsx

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { FunctionComponent, ReactElement } from 'react';
+import { Props, propTypes } from './types';
+export { Props } from './types';
+
+const NxDonutChart: FunctionComponent<Props> =
+  function NxDonutChart(props): ReactElement<Props> {
+    const decimal = props.percent / 100;
+    const diameter = props.outerRadius * 2;
+    const strokeWidth = props.outerRadius - props.innerRadius;
+    const r = props.innerRadius + strokeWidth / 2;
+    const circumference = 2 * Math.PI * r;
+    const strokeDasharray = [decimal * circumference, (1.0 - decimal) * circumference];
+    const strokeDashoffset = (0.25 * circumference) - (1.0 - decimal) * circumference;
+    return (
+      <svg width={diameter} height={diameter}>
```

Comment:
> Rather than setting width and height here, you can leave that to CSS and use a viewBox to create an arbitrary (and more convenient) "user coordinate space". For instance, `viewBox="-1 -1 2 2" would give you a coordinate space with (0,0) in the center and the edges 1 unit away from that center. This would likely simplify your math above, and would allow the on-screen size of the chart to be controlled in css as it ought to be.

Additionally, if you center around (0,0), it makes it relatively easy to use hard-coded transforms to get the start/finish of the arc where you want them rather than having to calculate the dashoffset.

Reply:
> Additionally, if you center around (0,0), it makes it relatively easy to use hard-coded transforms to get the start/finish of the arc where you want them rather than having to calculate the dashoffset.

These are the transforms I'm thinking of: `transform: rotate(-90deg) scaleY(-1);`

Reply:
> use a `viewBox`

> These are the transforms I'm thinking of: `transform: rotate(-90deg) scaleY(-1);`

Used a `viewBox` and transforms in https://github.com/sonatype/sonatype-react-shared-components/pull/172/commits/5a3e4fd020df42fb5582f6d3dd79160c5e40120b seem to work nicely